Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces;
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Managers
+{
+    /// <summary>
+    /// The Mixed Reality manager is responsible for coordinating the operation of the Mixed Reality Toolkit
+    /// It provides a service registry for all active managers that are used within a project as well as providing the active configuration profile for the project
+    /// The Profile can be swapped out at any time to meet the needs of your project.
+    /// </summary>
+    public class MixedRealityManager : Utilities.Singleton<MixedRealityManager>
+    {
+        #region Mixed Reality Manager Profile configuration
+
+        /// <summary>
+        /// The active profile of the Mixed Reality Manager which controls which components are active and their initial configuration.
+        /// *Note configuration is used on project initialization or replacement, changes to properties while it is running has no effect.
+        /// </summary>
+        [SerializeField]
+        [Tooltip("The current active configuration for the Mixed Reality project")]
+        private MixedRealityConfigurationProfile activeProfile;
+
+        /// <summary>
+        /// The public property of the Active Profile, ensuring events are raised on the change of the configuration
+        /// </summary>
+        public MixedRealityConfigurationProfile ActiveProfile
+        {
+            get { return activeProfile; }
+            set { activeProfile = value; if (ProfileUpdateEvent != null) ProfileUpdateEvent.Invoke(); }
+        }
+
+        #endregion
+
+        #region Active SDK components
+
+        /// <summary>
+        /// The Active Controllers property lists all the controllers detected by the Mixed Reality manager on startup
+        /// </summary>
+        [SerializeField]
+        [Tooltip("The collection of currently active / detected controllers")]
+        private Controller[] activeControllers;
+
+        /// <summary>
+        /// The Active Headset property maintains the Headsets/SDK detected by the Mixed Reality manager on startup
+        /// </summary>
+        [SerializeField]
+        [Tooltip("The currently active / detected Headset or SDK")]
+        private Headset activeHeadset;
+
+        #endregion
+
+        #region Mixed Reality Manager Events
+
+        /// <summary>
+        /// Delegate for the Initialize event
+        /// </summary>
+        public delegate void InitializeHandler();
+        /// <summary>
+        /// Initialize event called after all managers have been registered. Informs all managers to initialize (Awake)
+        /// </summary>
+        public event InitializeHandler InitializeEvent;
```

Comment:
Instead of subscribing to events, let's just call the methods directly.

Reply:
OK, THat is certainly another way.  Having the methods defined in the interface and virtuals in the base.  Fair enough.

Reply:
Agreed